### PR TITLE
IGAPP-619: Sanitize endpoints

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: api-client/package.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
...
   "dependencies": {
     "entities": "^2.0.0",
+    "isomorphic-dompurify": "^0.14.0",
     "js-md5": "^0.7.3",
     "lodash": "^4.17.19",
     "moment": "2.29.1",
     "moment-timezone": "^0.5",
     "normalize-path": "^3.0.0",
-    "react": "16.13.1",
-    "sanitize-html-react": "^1.13"
```
Comment (Taggotty) [inline api-client/package.json:26]: now you can also remove `src/@types/sanitize-html-react.ts`

File: api-client/src/mapping/mapCategoryJson.ts [hunk 2]
```diff
@@ -3,18 +3,14 @@ import CategoryModel from '../models/CategoryModel'
...
   new CategoryModel({
     root: false,
     path: normalizePath(json.path),
     title: json.title,
-    content: sanitizeHtml(json.content, {
-      allowedSchemes: ['http', 'https', 'data', 'tel', 'mailto'],
-      allowedTags: false,
-      allowedAttributes: false
-    }),
+    content: sanitize(json.content),
```
Comment (Taggotty) [inline api-client/src/mapping/mapCategoryJson.ts:13]: was there a reason why we didn't support ftp before?

**Review by Taggotty — APPROVED** [review]

File: api-client/src/mapping/mapCategoryJson.ts (hunk 2 above)
Comment (klinzo) [reply api-client/src/mapping/mapCategoryJson.ts:13]: @maxammann

File: api-client/package.json (hunk 1 above)
Comment (klinzo) [reply api-client/package.json:26]: I think I did remove them, didn't I?

Comment (Taggotty) [reply api-client/package.json:26]: oh, yes you did. missed that.

File: api-client/package.json [hunk 3]
```diff
@@ -17,13 +17,13 @@
   },
   "dependencies": {
     "entities": "^2.0.0",
+    "isomorphic-dompurify": "^0.14.0",
```
Comment (maxammann) [inline api-client/package.json:20]: Why aren't we using dompurify directly?

File: api-client/src/mapping/mapCategoryJson.ts (hunk 2 above)
Comment (maxammann) [reply api-client/src/mapping/mapCategoryJson.ts:13]: I dont think so, just no usecase. FTP is also not supported in most browsers afaik.

**Review by maxammann — APPROVED** [review]

File: api-client/package.json (hunk 3 above)
Comment (klinzo) [reply api-client/package.json:20]: So I understood @Taggotty wrong, she just said that maybe `dompurify` does not work on native but I thought she checked and it does not work. However, it does work, so we'll use `dompurify` directly.

**Pull request merged**